### ISSUE TITLE
document builtins.importNative

### ISSUE THREAD

**Issue by lolbinarycat** [NixOS/nix#10366]
## Problem

this is quite powerful, a possible attack vector, and also completely undocumented

## Proposal

document it

## Checklist

- [x] checked [latest Nix manual] \([source])
- [ ] checked [open documentation issues and pull requests] for possible duplicates

[latest Nix manual]: https://nixos.org/manual/nix/unstable/
[source]: https://github.com/NixOS/nix/tree/master/doc/manual/src
[open documentation issues and pull requests]: https://github.com/NixOS/nix/labels/documentation

## Priorities

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by adisbladis** [comment]
Rather than documenting it I'd recommend a removal.

**Comment by lolbinarycat** [comment]
how much of nix should be ripped apart for that?

there seems to be an entire framework for supporting foreign types.

**Comment by thufschmitt** [comment]
> Rather than documenting it I'd recommend a removal.

Agreed. This is quite a bit of convoluted thing just to be able to not run a fork of Nix. `builtins.exec` _might_ have use-cases, but `importNative` is a net negative in my opinion.

**Comment by fricklerhandwerk** [comment]
Closed by #10803 and #10810. We can still, independently, discuss the merits of that feature.
